### PR TITLE
Document the ownCloud App for Splunk

### PR DESCRIPTION
**Pull request by pmaier1**
@mmattel @EParzefall can one of you take over? Let me know if there are questions.

### CODE REVIEW

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+By aggregating, evaluating and visualizing the data provided by ownCloud, the ownCloud App for Splunk allows service providers to gain insights into how their ownCloud platform is used and adopted (e.g., user, storage and sharing growth). Automatically gathering and processing ownCloud data enables to build up a continuous reporting tool for stakeholders. For auditing purposes the app provides very fine-grained and flexible tools that allow tracing actions by user, by operation or even by a single file, and more.
+
+The app makes all relevant ownCloud data available in Splunk. With that, the dashboards and tools can easily be extended or modified. With just a few clicks they can be adapted to specific needs using the filtering and visualization features provided by Splunk.
+
+== Prerequisites
+To get set up with the _ownCloud App for Splunk_ there is a number of prerequisites that have to be fulfilled.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:12]:
```suggestion
To get set up with the _ownCloud App for Splunk_ there are a number of prerequisites that have to be fulfilled.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+* ownCloud Server has a minimum version of 10.5
+* Splunk has a minimum version of 7.2
+* The https://marketplace.owncloud.com/apps/metrics[Metrics App] is installed, configured and enabled on ownCloud Server
+* The https://marketplace.owncloud.com/apps/admin_audit[Auditing App] is installed, configured and enabled on ownCloud Server
+* Both components of the _ownCloud App for Splunk_ are installed and configured
+
+== Setup & Configuration
+=== ownCloud
+
+1. Install and set up the https://marketplace.owncloud.com/apps/admin_audit[Auditing App] as https://doc.owncloud.com/server/admin_manual/enterprise/logging/admin_audit.html[documented]. Especially take care for the log file paths (owncloud.log / admin_audit.log) as those will be required in the Splunk configuration below.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:23]:
```suggestion
1. Install and set up the https://marketplace.owncloud.com/apps/admin_audit[Auditing App] as https://doc.owncloud.com/server/admin_manual/enterprise/logging/admin_audit.html[documented]. Especially take care of the log file paths (owncloud.log / admin_audit.log) as those will be required in the Splunk configuration below.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+2. Install and set up the https://marketplace.owncloud.com/apps/metrics[Metrics App] as https://doc.owncloud.com/server/latest/admin_manual/enterprise/reporting/metrics.html[documented]. Especially take care of setting the Metrics API key as it will be required in the Splunk configuration below.
+
+=== Splunk
+
+The _ownCloud App for Splunk_ consists of two components that have to be installed and configured in Splunk.
+
+* The https://splunkbase.splunk.com/app/5502/[_ownCloud Add-on for Splunk_] gathers and stores the ownCloud data in Splunk.
+* The https://splunkbase.splunk.com/app/5503/[_ownCloud App for Splunk_] adds dashboards and other functionalities to the Splunk web interface.
+
+Both can be installed from the Splunkbase app store. Below you find the necessary configuration to get started.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:34]:
```suggestion
Both can be installed from the Splunkbase app store. You will find the necessary initial configuration below.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+2. Install a Splunk Universal Forwarder on your ownCloud host. For further help, see https://docs.splunk.com/Documentation/Forwarder/latest/Forwarder/Installanixuniversalforwarder.
+
+3. Install the _ownCloud Add-on for Splunk_
+
+* If you're using a standalone Splunk instance, you have to install the _ownCloud Add-on for Splunk_.
+* If you're using a distributed Splunk installation, it depends on your setup:
+  * Search Heads: Installation of the _ownCloud Add-on for Splunk_ is required
+  * Indexers: Installation of the _ownCloud Add-on for Splunk_ is conditional. It is not required if you use Heavy Forwarders to collect data. It is required if you use Universal Forwarders to collect data.
+  * Universal or Heavy Forwarders: Installation of the _ownCloud Add-on for Splunk_ required. In addition it is required to enable data and scripted input as described below.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:54]:
```suggestion
  * Universal or Heavy Forwarders: Installation of the _ownCloud Add-on for Splunk_ required. In addition data and scripted input must be enabled as described below.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 5]
```diff
@@ -0,0 +1,77 @@
+= ownCloud App for Splunk
+
+== Introduction
+
+The https://splunkbase.splunk.com/app/5503/[ownCloud App for Splunk] provides a sophisticated reporting and auditing tool for ownCloud service operators. It makes use of both ownCloud's technical logs (owncloud.log) and audit logs, and the ownCloud Metrics API to provide insights. It shows information about users as well as storage and sharing usage across the instance and per user. It also makes audit evaluations quicker and more efficient. The app configures Splunk to retrieve and store the data and to provide visualizations, log filtering tools and pre-defined alerts for certain events. 
+
+By aggregating, evaluating and visualizing the data provided by ownCloud, the ownCloud App for Splunk allows service providers to gain insights into how their ownCloud platform is used and adopted (e.g., user, storage and sharing growth). Automatically gathering and processing ownCloud data enables to build up a continuous reporting tool for stakeholders. For auditing purposes the app provides very fine-grained and flexible tools that allow tracing actions by user, by operation or even by a single file, and more.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:7]:
```suggestion
By aggregating, evaluating and visualizing the data provided by ownCloud, the ownCloud App for Splunk allows service providers to gain insights into how their ownCloud platform is used and adopted (e.g., user, storage and sharing growth). Automatically gathering and processing ownCloud data enables a continuous reporting tool to be built up for stakeholders. For auditing purposes the app provides very fine-grained and flexible tools that allow tracing actions by user, by operation or even by a single file, and more.
```

**Review by phil-davis — COMMENTED** [review]
Just a few little suggestions for English wording.

**Comment by pmaier1** [comment]
Thanks, Phil :+1:

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+* If you're using a distributed Splunk installation, it depends on your setup:
+  * Search Heads: Installation of the _ownCloud Add-on for Splunk_ is required
+  * Indexers: Installation of the _ownCloud Add-on for Splunk_ is conditional. It is not required if you use Heavy Forwarders to collect data. It is required if you use Universal Forwarders to collect data.
+  * Universal or Heavy Forwarders: Installation of the _ownCloud Add-on for Splunk_ required. In addition data and scripted input must be enabled as described below.
+
+4. Enable data and scripted input with a configuration file
+
+On your Universal Forwarder or Heavy Forwarder instance, you must enable the inputs using the configuration files.
+
+1. Copy `$SPLUNK_HOME/etc/apps/TA_owncloud/default/inputs.conf.example` to `$SPLUNK_HOME/etc/apps/TA_owncloud/local` directory and rename the file to `inputs.conf`.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:60]:
```suggestion
. Copy `$SPLUNK_HOME/etc/apps/TA_owncloud/default/inputs.conf.example` to `$SPLUNK_HOME/etc/apps/TA_owncloud/local` directory and rename the file to `inputs.conf`.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+2. Open `$SPLUNK_HOME/etc/apps/TA_owncloud/local/inputs.conf` for editing.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:61]:
```suggestion
. Open `$SPLUNK_HOME/etc/apps/TA_owncloud/local/inputs.conf` for editing.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+3. Check all `index = owncloud` settings and change the index name if needed.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:62]:
```suggestion
. Check all `index = owncloud` settings and change the index name if needed.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+4. Check the ownCloud Logs locations (default: `/var/www/owncloud/data/`) and change them to the values you configured on the ownCloud Server.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:63]:
```suggestion
. Check the ownCloud Logs locations (default: `/var/www/owncloud/data/`) and change them to the values you configured on the ownCloud Server.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+4. Save the `$SPLUNK_HOME/etc/apps/TA_owncloud/local/inputs.conf` file.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:64]:
```suggestion
. Save the `$SPLUNK_HOME/etc/apps/TA_owncloud/local/inputs.conf` file.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+5. Copy `$SPLUNK_HOME/etc/apps/TA_owncloud/default/owncloud.conf.example` to `$SPLUNK_HOME/etc/apps/TA_owncloud/local` directory and rename the file to `owncloud.conf`.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:65]:
```suggestion
. Copy `$SPLUNK_HOME/etc/apps/TA_owncloud/default/owncloud.conf.example` to `$SPLUNK_HOME/etc/apps/TA_owncloud/local` directory and rename the file to `owncloud.conf`.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+6. Open `$SPLUNK_HOME/etc/apps/TA_owncloud/local/owncloud.conf` for editing.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:66]:
```suggestion
. Open `$SPLUNK_HOME/etc/apps/TA_owncloud/local/owncloud.conf` for editing.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+7. Change the `METRICSAPIKEY` setting to the Metrics API key value you configured on the ownCloud Server.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:67]:
```suggestion
. Change the `METRICSAPIKEY` setting to the Metrics API key value you configured on the ownCloud Server.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+8. Change the `API_HOST` setting to your ownCloud instance domain name or IP address. This value is used to query the Metrics API for data.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:68]:
```suggestion
. Change the `API_HOST` setting to your ownCloud instance domain name or IP address. This value is used to query the Metrics API for data.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+9. Save the `$SPLUNK_HOME/etc/apps/TA_owncloud/local/owncloud.conf` file.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:69]:
```suggestion
. Save the `$SPLUNK_HOME/etc/apps/TA_owncloud/local/owncloud.conf` file.
```

File: modules/admin_manual/pages/configuration/integration/splunk.adoc [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+10. Restart the Splunk instance.
```
Comment (EParzefall) [inline modules/admin_manual/pages/configuration/integration/splunk.adoc:70]:
```suggestion
. Restart the Splunk instance.
```

**Review by EParzefall — COMMENTED** [review]
Fixed layout and references according to ownCloud best practices and some style/content issues. @pmaier1  Please check and commit these changes, then I can do a build and check the layout before merging.

**Comment by pmaier1** [comment]
Thanks a lot @EParzefall :ok_hand:

**Review by EParzefall — APPROVED** [review]

**Pull request merged**